sdk/typescript/introspector/test/invoke.spec.ts

```diff
@@ -169,4 +169,80 @@ describe("Invoke typescript function", function () {
       { LogOutput: process.stderr }
     )
   })
+
+  describe("Should correctly invoke variadic functions", async function () {
+    this.timeout(60000)
+
+    type Case = {
+      [name: string]: { ctx: InvokeCtx; expected: string | number }
+    }
+
+    const cases: Case = {
+      "invoke full variadic string function": {
+        expected: "hello hello world",
+        ctx: {
+          parentName: "Variadic",
+          fnName: "fullVariadicStr",
+          parentArgs: {},
+          fnArgs: {
+            vars: ["hello", "world"],
+          },
+        },
+      },
+      "invoke variadic function with fixed first argument": {
+        expected: "hello hello+world",
+        ctx: {
+          parentName: "Variadic",
+          fnName: "semiVariadicStr",
+          parentArgs: {},
+          fnArgs: {
+            separator: "+",
+            vars: ["hello", "world"],
+          },
+        },
+      },
+      "invoke full variadic number function": {
+        expected: 3,
+        ctx: {
+          parentName: "Variadic",
+          fnName: "fullVariadicNum",
+          parentArgs: {},
+          fnArgs: {
+            vars: [1, 2],
+          },
+        },
+      },
+      "only invoke variadic function with fixed first argument": {
+        expected: 12,
+        ctx: {
+          parentName: "Variadic",
+          fnName: "semiVariadicNum",
+          parentArgs: {},
+          fnArgs: {
+            mul: 2,
+            vars: [1, 1, 1, 2, 1], // 6
+          },
+        },
+      },
+    }
+
+    for (const [name, { ctx, expected }] of Object.entries(cases)) {
+      it(name, async function () {
+        const files = await listFiles(`${rootDirectory}/variadic`)
+
+        // Load function
+        await load(files)
+
+        const scanResult = scan(files)
```

Comment:
This setup doesn't need to be repeated. You could use a [`before` hook](https://mochajs.org/#hooks):

```ts
let scanResult: ScanResult

before(function () {
    ...
    scanResult = scan(files)
})

it(..., function () {
    invoke(scanResult)
})
```

Reply:
I'll update the in a follow up PR